shared_test_utils/src/integration_test_utils/integration_test_utils.py

```diff
@@ -0,0 +1,495 @@
+    )
+
+
+async def deploy_prometheus_configurer(
+    ops_test,
+    channel="latest/stable",
+) -> None:
+    """Deploys prometheus-configurer-k8s charm.
+
+    Args:
+        ops_test (OpsTest)
+        channel (str): channel from which the charm will be deployed
+    """
+    await ops_test.model.deploy(
+        itest_const.PROMETHEUS_CONFIGURER_K8S_CHARM_NAME,
+        application_name=itest_const.PROMETHEUS_CONFIGURER_K8S_CHARM_NAME,
+        channel=channel,
+        trust=True,
+    )
+    await ops_test.model.add_relation(
+        relation1=itest_const.PROMETHEUS_CONFIGURER_K8S_CHARM_NAME,
+        relation2=f"{itest_const.PROMETHEUS_K8S_CHARM_NAME}:\
+            receive-remote-write",
+    )
+
+
+async def deploy_prometheus_k8s_operator(
+    ops_test,
+    channel="latest/stable",
+) -> None:
+    """Deploys prometheus-k8s charm.
+
+    Args:
+        ops_test (OpsTest)
+        channel (str): channel from which the charm will be deployed
+    """
+    await ops_test.model.deploy(
+        itest_const.PROMETHEUS_K8S_CHARM_NAME,
+        application_name=itest_const.PROMETHEUS_K8S_CHARM_NAME,
+        channel=channel,
+        trust=True,
+    )
+    await ops_test.model.add_relation(
+        relation1=f"{itest_const.PROMETHEUS_K8S_CHARM_NAME}:grafana-source",
+        relation2=itest_const.GRAFANA_K8S_CHARM_NAME,
+    )
+
+
+async def deploy_nms_magmalte(ops_test) -> None:
+    """Deploys nms-magmalte-operator charm.
+
+    Args:
+        ops_test (OpsTest)
+    """
+    magmalte_charm = find_charm(
+        "../nms-magmalte-operator",
+        itest_const.NMS_MAGMALTE_CHARM_FILE_NAME,
+    )
+
+    if not magmalte_charm:
+        magmalte_charm = await ops_test.build_charm(
+            "../nms-magmalte-operator/"
+        )  # noqa: E501
+    resources = {
+        f"{itest_const.NMS_MAGMALTE_CHARM_NAME}-image": itest_const.NMS_MAGMALTE_METADATA[  # noqa: E501
+            "resources"
+        ][
+            f"{itest_const.NMS_MAGMALTE_CHARM_NAME}-image"
+        ][
+            "upstream-source"
+        ],
+    }
+    await ops_test.model.deploy(
+        magmalte_charm,
+        resources=resources,
+        application_name=itest_const.NMS_MAGMALTE_APPLICATION_NAME,
+        trust=True,
+        series="jammy",
+    )
+    await ops_test.model.add_relation(
+        relation1=itest_const.NMS_MAGMALTE_APPLICATION_NAME,
+        relation2=f"{itest_const.DB_CHARM_NAME}:database",
+    )
+    await ops_test.model.add_relation(
+        relation1=itest_const.NMS_MAGMALTE_APPLICATION_NAME,
+        relation2=f"{itest_const.CERTIFIER_APPLICATION_NAME}:\
+            cert-admin-operator",
+    )
+    await ops_test.model.add_relation(
+        relation1=f"{itest_const.NMS_MAGMALTE_APPLICATION_NAME}:\
+            grafana-auth",
+        relation2=itest_const.GRAFANA_K8S_CHARM_NAME,
+    )
+
+
+async def deploy_prometheus_cache(ops_test, channel="latest/stable") -> None:
+    """Deploys "prometheus-edge-hub" charm.
```

Comment:
I would remove this just for consistency across docstrings.
```suggestion
    """Deploys prometheus-edge-hub charm.
```

Reply:
✅